app/services/imo_api_service.rb

```diff
@@ -0,0 +1,71 @@
+class ImoApiService
+  IMO_USERNAME = ENV["IMO_USERNAME"]
+  IMO_PASSWORD = ENV["IMO_PASSWORD"]
+  BASE_URL = "https://sgp.imo.im/api/simple/"
+
+  class ImoApiService::HTTPError < HTTP::Error
+  end
+
+  attr_reader :phone_number, :recipient_name, :locale
+
+  def initialize(phone_number:, recipient_name:, locale:)
```

Comment:
Lets pass in `patient` here, and let the service grab what it needs from the patient.

The API request is tightly coupled with the patient, so I think its best just to hand it the patient object and let the API grab whatever things it needs for the request.

Reply:
The API is only really tied to patients in the sense that we don't currently have a reason to send imo messages to non-patients. I was trying to keep this as generic as possible for that reason. I find it unlikely we will ever want to use imo with users, but this still seems like reasonable separation of concerns.